src/idlcxx/src/streamer_generator.c

```diff
@@ -532,6 +547,27 @@ void load_locals(context_t* ctx, bool* storage)
   ctx->key_funcs.sequenceentriespresent = storage[3];
 }
 
+bool has_keys(idl_type_spec_t* spec)
+{
+  //is struct
+  if (idl_is_struct(spec))
+  {
+    idl_member_t* mem = ((idl_struct_t*)spec)->members;
+    while (mem)
+    {
+      if (idl_is_masked(mem, IDL_KEY)) return true;
+      mem = (idl_member_t*)(mem->node.next);
```

Comment:
You could use `idl_next` here

Reply:
idl_next looks like a good replacement for these kinds of expressions, I will switch to idl_next